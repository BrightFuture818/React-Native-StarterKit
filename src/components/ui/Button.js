/**
 * Buttons
 *
     <Button text={'Server is down'} />
 *
 * React Native Starter App
 * https://github.com/mcnamee/react-native-starter-app
 */
import React, { Component, PropTypes } from 'react';
import { Alert } from 'react-native';
import { Button } from 'react-native-elements';

// Consts and Libs
import { AppColors, AppFonts, AppSizes } from '@theme/';

/* Component ==================================================================== */
class CustomButton extends Component {
  static propTypes = {
    small: PropTypes.bool,
    large: PropTypes.bool,
    outlined: PropTypes.bool,
    backgroundColor: PropTypes.string,
    icon: PropTypes.shape({
      name: PropTypes.string,
    }),
  }

  static defaultProps = {
    small: false,
    large: false,
    outlined: false,
    icon: {},
  }

  buttonProps = () => {
    // Defaults
    const props = {
      title: 'Coming Soon...',
      onPress: () => Alert.alert('Coming soon...'),
      backgroundColor: AppColors.brand.primary,
      color: '#fff',
      fontWeight: 'bold',
      fontFamily: AppFonts.base.family,
      fontSize: AppFonts.base.size,
      borderRadius: AppSizes.borderRadius,
      raised: true,
      buttonStyle: {
        padding: 12,
        marginLeft: 0,
        marginRight: 0,
      },
      ...this.props,
      small: false,
      large: false,
      icon: (this.props.icon && this.props.icon.name)
        ? {
          ...this.props.icon,
          size: 14,
        } : null,
    };

    // Overrides
    // Size
    if (this.props.small) {
      props.fontSize = 12;
      props.buttonStyle.padding = 8;

      if (props.icon && props.icon.name) {
        props.icon = {
          ...props.icon,
          size: 14,
        };
      }
    }
    if (this.props.large) {
      props.fontSize = 20;
      props.buttonStyle.padding = 15;

      if (props.icon && props.icon.name) {
        props.icon = {
          ...props.icon,
          size: 20,
        };
      }
    }

    // Outlined
    if (this.props.outlined) {
      props.raised = false;
      props.backgroundColor = this.props.backgroundColor || 'transparent';
      props.color = AppColors.brand.primary;
      props.buttonStyle.borderWidth = 1;
      props.buttonStyle.borderColor = AppColors.brand.primary;

      if (props.icon && props.icon.name) {
        props.icon = {
          ...props.icon,
          color: AppColors.brand.primary,
        };
      }
    }

    return props;
  }

  render = () => <Button {...this.buttonProps()} />;
}

/* Export Component ==================================================================== */
export default CustomButton;
