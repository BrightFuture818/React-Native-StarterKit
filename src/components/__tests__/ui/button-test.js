/**
 * Test to check if the component renders correctly
 */
/* global it expect */

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Button from '@ui/Button';

/**
 * Check if button renders correcly
 * and asserting it to the matching snapshot
 */
it('Button renders correcly', () => {
  const tree = renderer.create(
    <Button />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
