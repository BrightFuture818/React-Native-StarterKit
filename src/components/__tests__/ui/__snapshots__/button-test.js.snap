exports[`test Button renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#9E9E9E",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginLeft": 15,
          "marginRight": 15,
          "padding": 19,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        Object {
          "backgroundColor": "#0E4EF8",
        },
        Object {
          "borderRadius": 2,
        },
        Object {
          "shadowColor": "rgba(0,0,0, .4)",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
        },
        Object {
          "padding": 12,
        },
        Object {
          "marginLeft": 0,
          "marginRight": 0,
          "padding": 12,
        },
        undefined,
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": undefined,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "white",
              "fontSize": 20,
            },
            Object {
              "color": "#fff",
            },
            Object {
              "fontSize": 17.5,
            },
            Object {
              "fontSize": 14,
            },
            undefined,
            Object {
              "fontWeight": "bold",
            },
            Object {
              "fontFamily": "HelveticaNeue",
            },
          ],
        ]
      }>
      Coming Soon...
    </Text>
  </View>
</View>
`;
